libr/lang/p/python/anal.c

```diff
@@ -155,8 +157,8 @@ static int py_anal(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
 	if (py_anal_cb) {
 		memset(op, 0, sizeof (RAnalOp));
 		// anal(addr, buf) - returns size + dictionary (structure) for RAnalOp
-		PyObject *arglist = Py_BuildValue ("(i, "BYTES_FMT")", addr, buf, len);
-		PyObject *result = PyEval_CallObject (py_anal_cb, arglist);
+		PyObject *arglist = Py_BuildValue ("("BYTES_FMT")", buf, len, addr);
```

Comment:
Why did you change the order? According to book it is `addr` first, then `buf` second: https://radare.gitbooks.io/radare2book/content/plugins/python.html

Reply:
I changed it in the book as well to be compatible with the diassemble()-api (see https://github.com/radare/radare2book/pull/184). The problem with that line was, that it expects 4 parameters (i.e. BYTES_FMT evalutes to y#iK) but only gets 3, so it didn't work at all.

As far as I know, there's no plugin yet that uses this API.